include/matx/transforms/copy.h

```diff
@@ -69,11 +69,39 @@ namespace matx
       (out = in).run(exec);
     };
 
-  template <typename OutputTensor, typename InputTensor>
+  template <typename OutputTensor, typename InputTensor, std::enable_if_t<is_tensor_view_v<InputTensor>, bool> = true>
     __MATX_INLINE__ void copy(OutputTensor &out, const InputTensor &in,
         cudaStream_t stream = 0)
     {
-      MATX_NVTX_START("", matx::MATX_NVTX_LOG_API)      
+      MATX_NVTX_START("", matx::MATX_NVTX_LOG_API)
       matx::copy(out, in, cudaExecutor(stream));
-    };    
+    };
+
+  /**
+   * Make a deep copy of a view into a new tensor.
+   *
+   * Creates a new tensor of the same rank and dimension as the input view.
+   * Copies the data from the input view into the new tensor and returns it.
+   *
+   * @param in
+   *   Tensor to copy from
+   * @param exec
+   *   Executor to run copy in
+   * @return New tensor into which the input view has been copied
+   */
+  template <typename Tensor, typename Executor, std::enable_if_t<is_executor_t<Executor>(), bool> = true>
+  __MATX_INLINE__ Tensor copy(const Tensor &in, Executor exec)
+  {
+    MATX_NVTX_START("", matx::MATX_NVTX_LOG_API)
+    Tensor out({in.Shape()});
```

Comment:
What are the braces here for?

Reply:
Good question - I had gotten a build error early on without them and meant to return to that to investigate. It may have been something wrong in the test cases though, so I'll check again.

Reply:
I pushed an update to use Descriptor() rather than Shape()